test/integration/targets/consul/tasks/main.yml

```diff
@@ -0,0 +1,99 @@
+---
+- name: Install Consul and test
+
+  vars:
+    consul_version: '1.5.0'
+    consul_uri: https://releases.hashicorp.com/consul/{{ consul_version }}/consul_{{ consul_version }}_{{ ansible_system | lower }}_{{ consul_arch }}.zip
+    consul_cmd: '{{ local_temp_dir }}/consul'
+
+  block:
+    - name: register pyOpenSSL version
+      command: "{{ ansible_python.executable }} -c 'import OpenSSL; print(OpenSSL.__version__)'"
+      register: pyopenssl_version
+
+    - name: Install python-consul
+      pip:
+        name: python-consul
+      register: result
+      until: result is success
+
+    - name: Create a local temporary directory
+      tempfile:
+        state: directory
+      register: tempfile_result
```

Comment:
Use `output_dir` instead of creating your own temporary directory. Although the path is currently static, ansible-test will clean it up between each test run. Eventually that var will point at an actual temporary directory.